src/blocks/scratch3_motion.js

```diff
@@ -51,24 +52,29 @@ class Scratch3MotionBlocks {
         util.target.setXY(x, y);
     }
 
-    goTo (args, util) {
+    getTarget (TO, util) {
         let targetX = 0;
         let targetY = 0;
-        if (args.TO === '_mouse_') {
+        if (TO === '_mouse_') {
             targetX = util.ioQuery('mouse', 'getX');
             targetY = util.ioQuery('mouse', 'getY');
-        } else if (args.TO === '_random_') {
+        } else if (TO === '_random_') {
             const stageWidth = this.runtime.constructor.STAGE_WIDTH;
             const stageHeight = this.runtime.constructor.STAGE_HEIGHT;
             targetX = Math.round(stageWidth * (Math.random() - 0.5));
             targetY = Math.round(stageHeight * (Math.random() - 0.5));
         } else {
-            const goToTarget = this.runtime.getSpriteTargetByName(args.TO);
+            const goToTarget = this.runtime.getSpriteTargetByName(TO);
             if (!goToTarget) return;
             targetX = goToTarget.x;
             targetY = goToTarget.y;
         }
-        util.target.setXY(targetX, targetY);
+        return [targetX, targetY];
+    }
+
+    goTo (args, util) {
+        let targetXY = getTarget(args.TO, util);
+        util.target.setXY(targetXY[0], targetXY[1]);
```

Comment:
Hmm.. compare this with line 68, `if (!goToTarget) return;`. In that case, `targetXY` will be the value `undefined`, and so doing `targetXY[0]` (or `targetXY[1]`) will throw an error (since doing `undefined[0]` doesn't make sense).

This would practically be a problem if `goTo` is given some nonexistant sprite's name; for example "go to (join (nonexistant) (sprite))".

Probably the simple solution would just be to check if targetXY isn't undefined: `if (targetXY) { util.target.setXY(...) }`.

This is also probably a problem in the `glideTo` definition.